http-server/src/main/java/com/facebook/airlift/http/server/testing/TestingHttpServerModule.java

```diff
@@ -62,6 +64,11 @@ public void configure(Binder binder)
             }
         });
 
+        HttpServerConfig httpServerConfig = buildConfigObject(HttpServerConfig.class);
+        if (!httpServerConfig.isAuthEnabled()) {
```

Comment:
Why is this needed?

Reply:
Injecting `Authorizer` when authorization check is disabled is not necessary.

In [`HttpServerProvider`](https://github.com/prestodb/airlift/blob/master/http-server/src/main/java/com/facebook/airlift/http/server/HttpServerProvider.java), which uses setter to inject the optional value, so it’s fine to not bind `Authorizer`.
But in `TestingHttpServer`, which uses constructor, if the parameter authorizer is not explicitly bound, we’ll have an error.

Anyhow, `OptionalBinder` looks more like a solution. Thanks @ajaygeorge for sharing this idea.